<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
<meta name="generator" content="http://www.nongnu.org/elyxer/"/>
<meta name="create-date" content="2016-02-17"/>
<link rel="stylesheet" href="http://elyxer.nongnu.org/lyx.css" type="text/css" media="all"/>
<title>Estimating Drivers Cell State Transitions using Gene Regulatory Network Models</title>
</head>
<body>
<div id="globalWrapper">
<h1 class="title">
Estimating Drivers Cell State Transitions using Gene Regulatory Network Models<span class="unknown">\author</span>[1]Daniel Schlauch <span class="unknown">\author</span>[2,3]Kimberly Glass <span class="unknown">\author</span>[1,3]John Quackenbush <span class="unknown">\affil</span>[1]Department of Biostatistics and Computational Biology, Dana-Farber Cancer Institute and Department of Biostatistics, Harvard TH Chan School of Public Health, Boston, MA 02115 <span class="unknown">\affil</span>[2]Channing Division of Network Medicine, Brigham and Women’s Hospital, Boston, MA 02115 <span class="unknown">\affil</span>[3]Department of Medicine, Harvard Medical School, Boston, MA 02115
</h1>
<div class="abstract">
<p class="abstract-message">
Abstract
</p>
In the language of systems biology, the state of a cell can be represented by a gene regulatory network that characterizes the gene transcriptional processes that are active in that cell type. And transitions that occur in a wide range of biological processes, ranging from development to disease, can be thought of as transformation of the gene regulatory network from its initial state to its final state. Here we propose a regression-based generalization of the PANDA method for gene regulatory network inference for individual states, and a regression approach to modeling cell state transitions, identifying transcription factors that alter the network structure as cell states change. Our results demonstrate the ability to find strong signals which persists across studies and tissues of the same disease and which are not detectable with conventional gene expression analysis.
</div>
<h1 class="Section-">
<a class="toc" name="toc-Section--1"></a>Introduction
</h1>
<div class="Unindented">
One of the fundamental problems is biology is modeling the transition between biological states such as that which occurs during development or as a healthy tissue transforms into a disease state. While it is appealing to conceive of this process as deterministic, reflecting a change from one well-defined phenotype to another, in truth the situation is much more complex. Neither the initial nor the final phenotype is discrete, but each falls into a continuum of states, which, on average, captures features of that phenotype <b>[many citations, too general?]</b>. Indeed, within each tissue there are many, many cells, each of which is its own particular instance of that tissue—with unique patterns of gene expression and individual regulatory processes<span class="unknown">\cite</span>mohanty2014model,hu2013heterogeneity. The same is true when considering individuals as each healthy and disease state is unique to each member of a study population. One way to conceptualize the state transition problem is to imagine that each phenotype has a characteristic gene regulatory network and that there are a set of processes that are either activated or inactivated to transform the network in the initial state into that characterizing the final state. Identifying those changes could, in principle, help us to understand not only the processes that drive the state change, but also how one might intervene to either promote or inhibit such a transition.
</div>
<div class="Indented">
As our ability to generate large-scale integrative multi-omic datasets, such as RNA-Seq <span class="unknown">\cite</span>licatalosi2010rna and whole genome sequencing, has grown, there has been an increased interest in using those data to infer gene regulatory networks to model fundamental biological processes. While there have been many network inference (NI) methods published utilizing gene expression or a combination of gene expression<span class="unknown">\cite</span>margolin2006aracne, marbach2012wisdom, wille2004sparse, faith2007large, zhang2005general and complementary ’Omics data <span class="unknown">\cite</span>ernst2008semi,glass2013passing, hecker2009gene, each of which uses a different approach to estimating the “strength” of interactions between genes (or between transcription factors and their targets), they all suffer from the same fundamental limitation. Methods typically rely on estimating weights that represent a measure of evidence of an interaction between two genes and then setting a threshold to identify “real” (high confidence) edges. While setting edge confidence thresholds allows us to graphically represent networks and allows us to compare networks based on the presence or absence of edges, it ultimately requires that we discard information regarding those “weak” edges that fail to reach an arbitrary cutoff. Now, one could argue that discarding low significance edges is sensible as one common goal in network inference is to deduce a single, high confidence network model that represents a particular phenotype under study or, in some cases, a transition between phenotypes. An alternative view is that these edges contain estimates for gene coexpression that may be muted due to the heterogeneity of the biological sample and that networks are best described by edgeweights on a continuous scale. Furthermore, we argue that the inclusion of all network edgeweights in downstream analyses allows for the collection of systematic, though subtle, differences in coexpression which may otherwise have been washed out by extraneous noise.
</div>
<h1 class="Section-">
<a class="toc" name="toc-Section--2"></a>Modeling cell state transitions as a problem in gene regulatory network transition
</h1>
<div class="Unindented">
Cell state transitions—such as those that occur during development, or as healthy tissue transforms into a disease phenotype—are fundamental properties of biological systems. Understanding what drives these transitions, and modeling the processes, is one of the great open challenges in modern biology. One way to conceptualize the state transition problem is to imagine that each phenotype has its own characteristic gene regulatory network, and that there are a set of processes that are either activated or inactivated to transform the network in the initial state into that which characterizes the final state. Identifying those changes could, in principle, help us to understand not only the processes that drive the state change, but also how one might intervene to either promote or inhibit such a transition. The starting point for modeling cell state transitions is to model the initial and final cell states. One might imagine that the initial and final cell states consist of characteristic processes, some of which are shared (sometimes referred to as “housekeeping” functions) and others which are unique to the particular state. The way we understand these processes is that they are controlled by gene regulatory networks in which transcription factors (and other regulators) moderate the transcription of individual genes whose expression characterizes the state. One way to represent such processes is to draw a directed network graph, in which transcription factors and genes are nodes network in the network, and edges represent the regulatory interactions between transcription factors and their target genes that are active in, and characteristic of, a particular cellular state. One way of representing such a network, with interactions between m transcription factors and n target genes, is as a binary <span class="formula"><i>m</i> × <i>n</i></span> “adjacency matrix,” with 1’s representing active transcription factor-target interactions, and 0’s representing the lack of a transcription factor-target gene regulatory interaction. One can then think of a cell fate transition as the process that transforms the network in its initial state to its final state form, adding and deleting edges to remake the network that characterizes one phenotype into that which characterizes the other. Using the adjacency matrix formalism, one can think of this as a problem in linear algebra in which we attempt to find an <span class="formula"><i>m</i> × <i>m</i></span> “transition matrix” <span class="formula"><b>T</b></span>, subject to a set of constraints that approximates the conversion from the initial network’s adjacency matrix <b><span class="formula"><b>A</b></span></b> into the final network’s adjacency matrix <span class="formula"><b>B</b></span>, or <div class="formula">
<b>B</b> = <b>AT</b>
</div>

</div>
<div class="Indented">
<div class="float">
<a class="Label" name="Figure-1"> </a><div class="figure" style="max-width: 50%;">
<div class="center">
<img class="figure" src="figures/figure1_a.png" alt="figure figures/figure1_a.png" style="max-width: 794px; max-height: 595px;"/>
<div class="caption">
Figure 1 <b>Overview of the Transition Matrix problem.</b> Our approach seeks to find the <span class="formula"><i>TF</i> × <i>TF</i></span> matrix which best characterizes the transition in from controls to cases at the gene regulatory network level. First, gene expression samples are collected for each group, followed by gene regulatory network inference performed on cases and controls in parallel. Finally, the transition matrix is calculated, estimating the changes in TF targetting which best characterize the state change from the control group to the cases group.
</div>

</div>

</div>

</div>

</div>
<div class="Indented">
While it is appealing to conceive of this process as deterministic, reflecting a change from one well-defined phenotype to another, in truth the situation is much more complex. Neither the initial nor the final phenotype is discrete, but each falls into a continuum of states, which, on average, captures the features of that phenotype. Indeed, within each tissue there are many, many cells, each of which is its own particular instance of that tissue—with unique patterns of gene expression and individual regulatory processes. In the language of adjacency matrices, what this means is that rather than representing each state by a matrix with binary entries, what one should do is use a representation in which entries are continuous, representing the strength of the transcription factor-target gene interaction averaged over the collection of samples (or cells) representing each state. And consequently, the problem of estimating the transition matrix is generalized to solving <span class="formula"><b>B</b> = <b>AT</b> + <b>E</b></span>, where <b>E</b> is an <span class="formula"><i>m</i> × <i>n</i></span> error matrix representing unexplained by our estimated transition and minimized by our procedure. In this formalism, modeling the cell state transition is equivalent to estimating the appropriate transition matrix <span class="formula"><b>T</b></span> that maps how the transcription factor-target gene interactions are “rewired” between states. And one could hypothesize that the drivers of the cell state transition are those transcription factors that have the greatest change in the targets that they regulate.
</div>
<div class="Indented">
To generate our gene regulatory networks we are commonly faced with state-specific gene expression data along with general regulatory prior knowledge regarding suspected regulatory relationships. [BERE-TM (NAME TBD)] performs this step by applying a novel, computationally efficient method which uses each TF’s DNA sequence binding motifs as classifiers for all genes and identifies genes which are coexpressed with the set of suspected targets as well as removes suspected targets which do not conform to the TF targets’ general pattern of expression (see supplementary materials and methods). This step yields two <span class="formula"><i>m</i> × <i>n</i></span> gene regulatory matrices for cases and controls each representing estimates of the targetting patterns of the <span class="formula"><i>m</i></span> TFs onto the <span class="formula"><i>n</i></span> genes.
</div>
<div class="Indented">
Next, the problem is formulated as a regression problem whereby we solve for the <span class="formula"><i>m</i> × <i>m</i></span> transition matrix which best describes the transformation from controls to cases (see supplementary materials and methods). The <span class="formula"><i>i</i><sup><i>th</i></sup></span> column in the matrix can be loosely interpreted as being the optimal linear combination of columns in the controls adjacency matrix which predict the <span class="formula"><i>i</i><sup><i>th</i></sup></span> column in the cases adjacency matrix. Naturally, we expect the transition matrix between two identical states to approximate the identity matrix. Intuitively, it is the deviation from the identity matrix which is of biological interest with off diagonal mass at point <span class="formula"><span class="symbol">(</span><i>i</i>, <i>j</i><span class="symbol">)</span></span> indicating a regulatory alteration between cases and controls involving the <span class="formula"><i>i</i><sup><i>th</i></sup></span> and <span class="formula"><i>j</i><sup><i>th</i></sup></span> TF. Focusing specifically on individual TF behavioral modifications, one might simply be interested in the overall changes for each column. This can be obtained via the sum of off-diagonal &ldquo;mass&rdquo; representing the sum of systematic modifications estimated in transitioning from controls to cases. Furthermore, the statistical significance of these results is estimated by permuting the case-control labels <span class="formula"><i>n</i></span> times and rerunning the the analysis to determine the proportion of null results which are more extreme than the observed result (see supplementary materials and methods).
</div>
<div class="Indented">
In evaluating the state transitions, we recognize the limitations of current network inference methods to predict individual edgeweights with a high degree of confidence. It’s therefore of interest to combine measurements across sets of edgeweights in order to extract meaningful signal from a network perturbation. Effectively, we approached the problem as a dimension reduction problem with the goal of identifying high-influence, systematic regulatory network alterations rather than isolated independent events. There are many existing methods for reducing a high-dimensional matrices which may be applied to a gene regulatory adjacency matrix. Commonly, Principal Components Analysis (PCA) identifies eigenvectors which can reconstruct the greatest degree of variance from the original data. One drawback of this approach is the lack of interpret-ability of these vectors. Our transition matrix approach can be considered as a data reduction method which (1) reduces the dimensionality of the differences between two networks (2) preserves the intuitive interpretation of its vectors and (3) utilizes our expectation that meaningful network transitions will occur via biologically systematic alterations and not via random, independent edge alterations.
</div>
<h1 class="Section-">
<a class="toc" name="toc-Section--3"></a>[BERE-TM (NAME TBD)] finds significantly differentially involved TFs in COPD with strong concordance in independent datasets
</h1>
<div class="Unindented">
We applied our method to four case-control datasets for Chronic Obstructive Pulmonary Disease (COPD)- Evaluation of COPD Longitudinally to Identify Predictive Surrogate Endpoints (ECLIPSE)<span class="unknown">\cite</span>vestbo2008evaluation (Figure <span class="unknown">\ref</span>fig:ECLIPSE_results) , the COPDGene study <span class="unknown">\cite</span>pillai2009genome (Supplemental Data), Lung Genomics Research Consortium (LGRC) <span class="unknown">\cite</span>lgrc(Supplemental Data) and Lung Tissue Chronic Obstructive Pulmonary Disease (LTCOPD) <span class="unknown">\cite</span>ltcopd (Supplemental Data). Each of these studies consisted of gene expression assays obtained from patients with COPD and a set of smoker controls. The tissue used in the ECLIPSE and COPDGene study was peripheral blood mononuclear cell (PBMC), while lung tissue was sampled for LGRC and LTCOPD. It is therefore reassuring that agreement is more strongly achieved between studies of the same tissue origin than acros tissues. Each of the four studies was most closely correlated with studies of the same tissue. However it is quite notable that the we do see much of the same dTFI signal across studies involving different tissue types. Gene regulatory networks derived from gene expression data are notoriously difficult to replicate across studies and it is of great interest that we have identified suspected mechanisms which are correlated not only across studies but across tissues as well.
</div>
<div class="Indented">
We separately applied our BERE network inference approach on cases and controls and computed the transition matrix. Top significance hits for dTFI showed strong concordance between each of the datasets. Out of 166 TF used in this study, seven were among top 10 most differentially involved in both the ECLIPSE and COPDGene studies (Figure <span class="unknown">\ref</span>fig:compare). Furthermore, three of these seven TFs (GABPA, ELK4, ELK1) also appeared as significant in the LGRC results with FDR&lt;.01.
</div>
<div class="Indented">
<div class="float">
<a class="Label" name="Figure-2"> </a><div class="figure">
<b>A</b><b><img class="embedded" src="figures/SSODMplot_scaled.png" alt="figure figures/SSODMplot_scaled.png" style="width: 100%; max-width: 1728px; height: auto; max-height: 504px;"/>
</b><b>B</b><img class="embedded" src="figures/Transition plot.png" alt="figure figures/Transition plot.png" style="width: 50%; max-width: 1080px; height: auto; max-height: 1080px;"/>
<div class="caption">
Figure 2 <b>(A)</b> <b>Differential transcription factor involvement in ECLIPSE study.</b> Observed TFs undergoing state transition (red) scaled for each TF by the distribution under the randomization of case-control labels (blue). <b>(B) Driver network transitions</b>. Network transitions are depicted here with arrows indicating the flow of targetting patterns from one transcription factor to another. Edges are sized according to the magnitude of the transition and nodes (TFs) are sized by the overall dTFI for each TF. The gain of targetting features is indicated by the color blue while the loss of features is indicated by red.
</div>

</div>

</div>

</div>
<div class="Indented">
Overall, there was a strong correlation between results in the two PBMC studies <span class="formula"><span class="symbol">(</span><i>r</i><sub><i>s</i></sub> = 0.87<span class="symbol">)</span></span> and the two lung tissue studies <span class="formula"><span class="symbol">(</span><i>r</i><sub><i>s</i></sub> = 0.71<span class="symbol">)</span></span> , and a weaker correlation between cross-tissue results <span class="formula"><span class="symbol">(</span><span class="bar"><i>r</i></span><sub><i>s</i></sub> = 0.63<span class="symbol">)</span></span>.
</div>
<div class="Indented">
Of further interest is the particular TFs which are differentially observed across studies. For example, ERS1, the gene encoding Estrogen receptor alpha (ES-<span class="formula"><i>α</i></span>) is likely to behave differently in males compared with females. We found that this TF is significantly differentially involved in the COPDGene study - identified as the <span class="formula">4<i>th</i></span> most differentially involved TF, but not the ECLIPSE study. Accordingly, the COPDGene study had a greater gender disparity <span class="formula"><span class="symbol">(</span><span class="mbox">Odds Ratio</span> = 1.126<span class="symbol">)</span></span> than the ECLIPSE study <span class="formula"><span class="symbol">(</span><span class="mbox">Odds Ratio</span> = 1.067<span class="symbol">)</span></span>.
</div>
<div class="Indented">
<div class="float">
<a class="Label" name="Figure-3"> </a><div class="figure">
<b>A</b><img class="embedded" src="figures/Magnitude_comparison_same_tissue.png" alt="figure figures/Magnitude_comparison_same_tissue.png" style="width: 60%; max-width: 1152px; height: auto; max-height: 648px;"/>
<b>B</b><img class="embedded" src="figures/Magnitude_comparison_diff_tissue.png" alt="figure figures/Magnitude_comparison_diff_tissue.png" style="width: 60%; max-width: 1152px; height: auto; max-height: 864px;"/>
<b>C</b><img class="embedded" src="figures/Venn.png" alt="figure figures/Venn.png" style="width: 40%; max-width: 3000px; height: auto; max-height: 2000px;"/>
<div class="caption">
Figure 3 <b>Comparison of dTFI for 166 TFs across four independent studies in two tissues</b>. ECLIPSE and COPGene data were obtained via PBMC and LGRC and LTCOPD were obtained via lung tissue. Results for two studies with gene expression data obtained from the same source (<b>A</b>), PBMC (left) and lung tissue (right) show strong concordance with a high degree of agreement, particularly for top hits. Spearman correlation for each of these tissue types was <span class="formula"><i>r</i><sub><i>s</i></sub> = .87</span> and <span class="formula"><i>r</i><sub><i>s</i></sub> = .71</span>, respectively. Correlations for across tissue study comparisons (<b>B</b>)<b> </b>demonstrated a weaker, but still measureable agreement. Each of the four studies was most consistent with the study of the same tissue type as its own. (<b>C</b>) Differentially involved transcription factors were found in agreement across datasets. Five TFs were discovered in the top 20 in each of the four studies. An additional 21 TFs were found in the top 20 TFs of at least two studies.
</div>

</div>

</div>

</div>
<div class="Indented">
Notably in the LGRC dataset, we discovered a differential targetting pattern involving the TFs RFX1 and RFX2. Both of these transcription factors were highly statistically significant (FDR&lt;.0001) and ranked as the top two results in the LGRC study. However, their signal was muted in the ECLIPSE and COPDGene studies, neither of which identified these transcription factors as drivers of the Smoker Control to COPD transition. There are many possible explanations for this result, but it is reasonable to speculate that the dramatically different result is due to the fact that the tissue of origin for the LGRC differs from the ECLIPSE and COPDGene tissue.
</div>
<div class="Indented">
<div class="float">
<a class="Label" name="Figure-4"> </a><div class="figure" style="max-width: 50%;">
<b><img class="figure" src="figures/dTFI vs LIMMA.png" alt="figure figures/dTFI vs LIMMA.png" style="max-width: 648px; max-height: 576px;"/>
</b><div class="caption">
Figure 4 <b>Comparison of significant results for TF differential involvement vs differential expression in ECLIPSE.</b> TFs which are differentially involved are not necessarily differentially expressed, and vice versa. Many TFs can be observed which have significantly different targetting patterns, but which are not statistically significantly differentially expressed. This suggests that our method finds transcription factors which are differentially affected at a post-transcription stage. 
</div>

</div>

</div>

</div>
<div class="Indented">
The top hits which are most consistent across studies have been implicated in independent studies for the development of COPD. Two of the top 3 hits, NRF1 and GABPA have been implicated in a mitochondrial mechanism for disease progression <span class="unknown">\cite</span>cloonan2016mitochondrial. Interestingly, the majority of the TFs identified as differentially involved do not exhibit significant differential gene expression. This suggests that for these proteins, their role in the disease may not occur until the post-transcription stage. It also suggests that conventional gene expression analysis is insufficient for identifying many of the TF drivers of disease.
</div>
<div class="Indented">
<h1 class="biblio">
References
</h1>

</div>

<hr class="footer"/>
<div class="footer" id="generated-by">
Document generated by <a href="http://elyxer.nongnu.org/">eLyXer 1.2.5 (2013-03-10)</a> on <span class="create-date">2016-02-17T11:49:27.782583</span>
</div>
</div>
</body>
</html>
