%!TEX root = ../dissertation.tex
% the abstract

The explosion of data arising from advances in high throughput sequencing has allowed scientists to study genomics in far greater detail.  However, this high resolution picture of cells often makes it difficult to see the higher level functions, features and structure in the biology that lead to phenotypic outcomes.  It would be simple if all phenotypes could be explained with a simple additive model consisting of the measured gene expression or variants in an organism.  However, it has become increasingly clear in the past 15 years that the drivers of fundamental biological processes involve complex systems of interactions between numerous genomic components.  These interactions are often described using network models, with nodes set as genomic features such as genes or proteins and edges inferred using one or more types of high throughput data.  Understanding how these networks should be constructed is an active field involving many types of interpretations of edges and methods for inference.  

With these advancements come new challenges in statistics.  The biological complexity of sample is not fully captured by a single snap shot of the sample's genome or gene expression profile.  Unknown and unmeasured structure exists at every genomic level in living organisms and has the potential to bias associations by confounding the relationship between genomic features and phenotypes.  Significant work has been published which attempts to identify and remove the impact of this unwanted variation, but subtle effects will continue to remain.  For example, EIGENSTRAT is a method for identifying hidden population structure along continuous axes in statistical genetics.  Left uncontrolled, this structure may cause spurious correlations in association studies.  Similarly, ComBat and Surrogate Variable Analysis are two methods in gene expression analysis which control for unwanted variation by removing the impact of confounding due to batch effect.  As the field advances, new applications and new technologies call for methods which improve on these tools and utilize all available information to provide better estimates.  Two of the chapters presented here deal with problems of identifying and removing unwanted structure in data.  Chapter 1 addresses a previously undocumented problem in the topic of gene coexpression by identifying and controlling for batch effect at the covariance level- that is confounding where coexpression is induced by a sample's membership in a batch.  Chapter 2 exploits the increased informativeness of rare variants to provide a higher resolution picture of population structure.  

